app/workers/classification_worker.rb

```diff
@@ -5,16 +5,28 @@ class ClassificationWorker
 
   sidekiq_options queue: :high
 
+  attr_reader :classification, :action
+
   def perform(id, action)
-    classification_lifecycle = ClassificationLifecycle.new(Classification.find(id))
+    @classification = Classification.find(id)
+    return if classification.lifecycled_at
```

Comment:
Won't this also return in case of `action == :update` (in which case it's been lifecycled before).